### ISSUE TITLE
Ability to reduce number of group badges when there are unassigned and unpaid badges left

### ISSUE THREAD

**Issue by ShevaDas** [magfest/ubersystem#2310]
This is possibly working as designed/a bad idea to implement, but:

If there is a group with 4 badges, 2 of which are unassigned, and the group has only paid for the 2 assigned badges, I would expect the dropdown where you can select the number of badges in the group to let me reduce the number from 4 to 2 and automatically remove the 2 unassigned badges.

**Comment by kitsuta** [comment]
Actually I think this is a bug, not a missing feature. You should be able to reduce badges down to the number of assigned badges -- I'm not sure why that's not working right now. I did look into it at the start when someone mentioned that they couldn't do it for a group, but I couldn't solve it, which means it is probably not a Newcomer issue. @EliAndrewC

**Comment by EliAndrewC** [comment]
Fair enough!  I agree it's a bug if it's not working.  I've added it to the onsite sprint so that we can at least look into it, and I was hopeful that it would turn out to be an easy fix, like an if-statement we added by mistake that could just be removed.  Probably not though :/

**Comment by khyperia** [comment]
I just tried to work on this, and was unable to reproduce the issue. Steps I took:

* Add a group with 4 members, assign 2 of them
* Pay enough for 2 members (using the test site, that's $80, while the page is requesting $160 for all 4 members) - not sure if this is needed, but I figured it's more accurate to a real-world situation.
* Use the count dropdown to go from 4->2 group members

It successfully removed the two unassigned badges.

I also tried with the group paying $40 and $120.

Could you give more info, or let me know if I read the issue wrong? Might be useful to know a real group with this issue in the live server, but I'm guessing that's long been forgotten. Also, what was the check-in status of the members already in? There's only a small chance that will make a difference, but it could be useful.

**Comment by kitsuta** [comment]
@khyperia I also could never reproduce the issue on my local copy, so it's likely that we're just missing some key element here. It's even possible this was somehow related to the bug where unassigned badges were being removed upon people checking in. I do know of one of the groups on the live server where this happened, but it looks like @ShevaDas was able to reduce their badges, so I'm confused.

**Comment by ShevaDas** [comment]
@kitsuta I believe I did that by going into the badges and deleting them individually. Wish I could remember what the error message was that sparked this thread originally...

**Comment by kitsuta** [comment]
It was this:
`You cannot reduce the number of badges for a group to below the number of assigned badges`

This is from `assign_badges` in models.py, but there's no obvious problem with that code. Without much more info, we may have to close this issue for now.

I'll send @khyperia the link to the group (we try to avoid talking about specific prod data on GitHub). The only thing I can think of is that it's a MIVS group, which means it was created automatically by the system and it has two 'free' badges.

**Comment by kitsuta** [comment]
Update! I was testing a different bug when I found what may have caused this.

![image](https://user-images.githubusercontent.com/7198215/28412713-0824b97e-6d13-11e7-8624-adaf0329cace.png)

As you can see, there is an unassigned badge there but I can't remove it. I _suspect_ that the reason I can't remove it is that it's not a "paid by group" badge. I checked the group link I had from earlier, and one of the badge in that group is also marked "doesn't need to" instead of "paid by group."

If that's the cause of this, then the problem is much simpler.
